pyincore/models/hazard/tornado.py

```diff
@@ -0,0 +1,185 @@
+            return hazard_service.post_tornado_hazard_values(self.id, payload, **kwargs)
+        else:
+            if self.tornado_type == "dataset":
+                return self.calculate_wind_speed_uniform_random_dist(payload)
+            else:
+                raise ValueError("Local Tornado type \"" + self.tornado_type + "\" is not supported yet.")
+
+    def calculate_wind_speed_uniform_random_dist(self, payload):
+        """ Read local hazard values from shapefile dataset
+
+                Args:
+                    payload (list):
+                Returns:
+                    obj: Hazard values.
+
+                """
+
+        response = []
+
+        # match demand types with shapefile file
+        for req in payload:
+            hazard_values = []
+            for index, req_demand_type in enumerate(req["demands"]):
+                match = False
+                for hazard_dataset in self.hazardDatasets:
+                    if hazard_dataset.dataset is None or not isinstance(hazard_dataset.dataset, Dataset):
+                        raise Exception("Hazard dataset is not properly attached to the hazard object.")
+
+                    # find matching raster file (Dataset) to read value from
+                    if req_demand_type.lower() == hazard_dataset.demand_type.lower():
+                        hazard_df = hazard_dataset.dataset.get_dataframe_from_shapefile()
+                        ef_box = -1
+                        len_fraction = 0.0
+                        x = float(req["loc"].split(",")[0])
+                        y = float(req["loc"].split(",")[1])
+                        location = Point(x, y)
+
+                        for _, feature in hazard_df.iterrows():
+                            polygon = feature['geometry']
+                            if location.within(polygon):
+                                ef_rating = feature[self.EF_RATING_FIELD]
+                                ef_box = Tornado.get_ef_rating(ef_rating)
+                                dist_pt = tornadoPath.distance(location)
+                                box_width = polygon.bounds[2] - polygon.bounds[0]
+                                len_fraction = dist_pt / box_width
+                                break
+
+                        if ef_box < 0:
+                            return None
+
+                        if ef_box == 5:  # EF5
+                            bottom_speed = self.EF_WIND_SPEED[ef_box]
+                            top_speed = self.MAX_WIND_SPEED
+                        else:
+                            bottom_speed = self.EF_WIND_SPEED[ef_box]
+                            top_speed = self.EF_WIND_SPEED[ef_box + 1]
+
+                        # Either generate random wind speed
+                        if True:
+                            random.seed(self.SEED)
+                            raw_wind_speed = random.uniform(bottom_speed, top_speed)
```

Comment:
It looks like this method has both linear interpolation and the uniform random, should all that extra code get removed?

Reply:
Ah I commit my experiments to this PR... will remove those shortly

Reply:
Done. Could you double check one more time.. my brain is not super working....